internal/callbacker/sender_test.go

```diff
@@ -0,0 +1,161 @@
+package callbacker_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/bitcoin-sv/arc/internal/callbacker"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCallbackSender_Send(t *testing.T) {
+	tests := []struct {
+		name            string
+		responseStatus  int
+		expectedSuccess bool
+		expectedRetries int
+	}{
+		{
+			name:            "success - callback sent",
+			responseStatus:  http.StatusOK,
+			expectedSuccess: true,
+			expectedRetries: 1,
+		},
```

Comment:
Add another case where `responseStatus` is an error e.g. `http.StatusInternalServerError`